lsp-lens.el

```diff
@@ -145,18 +145,11 @@ BUFFER-MODIFIED? determines whether the buffer is modified or not."
 
 (defun lsp-lens--create-interactive-command (command?)
   "Create an interactive COMMAND? for the lens."
-  (let ((server-id (->> (lsp-workspaces)
-                        (cl-first)
-                        (or lsp--cur-workspace)
-                        (lsp--workspace-client)
-                        (lsp--client-server-id))))
-    (if (functionp (lsp:command-command command?))
-        (lsp:command-command command?)
-      (lambda ()
-        (interactive)
-        (lsp-execute-command server-id
-                             (intern (lsp:command-command command?))
-                             (lsp:command-arguments? command?))))))
+  (if (functionp (lsp:command-command command?))
```

Comment:
Can `command?` be `nil` here?
If so, don't you need to check before passing it to `lsp--execute-command`?

Reply:
I'd guess no, because it wasn't possible previously (this wouldn't be a regression caused by this PR):
```emacs-lisp
    (if (functionp (lsp:command-command command?))
        (lsp:command-command command?)
      (lambda ()
        (interactive)
        (lsp-execute-command server-id
                             (intern (lsp:command-command command?))
                             (lsp:command-arguments? command?))))
; (if (functionp (lsp:command-command nil)) ...) -> (functionp nil) -> nil
```
This means that the control flow passed to the other branch, creating a lambda that would try to `intern` nil, which errors.

```emacs-lisp
(lsp:command-command nil) ; => nil
(intern nil) ; => error
```

Reply:
This function is called on line 190 in master (without this PR), and only lenses that have a command are passed to it (`-filter #'lsp:command-command`), so I guess command? cannot be nil. I'll note that in the docstring.

Reply:
Fixed.

Reply:
The reason I'm asking is because of parameter name `command?` which indicates it can be `nil`, we should change the parameter name too.

Reply:
I wouldn't do that, since it is an Elisp-level change, meaning that this entire thing has to be manually tested again, for something this trivial. See [this comment by @yyoncho](https://github.com/emacs-lsp/lsp-mode/pull/2317#discussion_r520169506). Noting that in the docstring is good enough, IMHO.

Also, the name isn't that bad: `command?` also corresponds to the naming of the `CodeLens` and `CodeAction` `:command?` fields, indicating that this function expects them to be passed in.